src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/globalmodule.cpp

```diff
@@ -82,15 +87,16 @@ ASPNET_CORE_GLOBAL_MODULE::OnGlobalConfigurationChange(
 
     // Test for an error.
     if (nullptr != pwszChangePath &&
-        _wcsicmp(pwszChangePath, L"MACHINE") != 0 &&
-        _wcsicmp(pwszChangePath, L"MACHINE/WEBROOT") != 0)
-    {
+        _wcsicmp(pwszChangePath, L"MACHINE/WEBROOT") > 0 &&
```

Comment:
If this is a prefix check shouldn't it be e.g.

```C++
_wcsnicmp(pwszChangePath, L"MACHINE/WEBROOT", wcslen(L"MACHINE/WEBROOT")) == 0
```

Why the >0 check?

Reply:
Also, I'm curious why the previous code looked like it did if it was in fact intending to do a prefix check.

Reply:
> Also, I'm curious why the previous code looked like it did if it was in fact intending to do a prefix check.

That's fair, it is written oddly. I haven't seen any examples of a non "MACHINE/WEBROOT/..." value which is why I assumed it was a prefix check.

Your proposed code would be an exact check not a prefix check. The `> 0` I'm using checks that it matches and contains more characters.

Reply:
Actually, the `wcsicmp` you're using will return a positive value for anything that's lexicographically greater, so even `MACHINE/ZZZ` will pass. It doesn't actually do a prefix check.

The wcsnicmp approach I suggested _would_ be a prefix check since it also specifies the number of characters to compare.

Reply:
Changed to just check for empty path